lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,123 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, captions=[], *, fmt="%g", position="edge",
```

Comment:
Can you change the default value to an empty tuple or `None`?  The list literal will be evaluated at method definition time so if you mutate it it will propagate between function calls. For example:


```python
def action_at_a_distance(*, inp=[]):
    inp.append(len(inp))
    return inp
```

```python
In [2]: action_at_a_distance()                                                                                                                                      
Out[2]: [0]

In [3]: action_at_a_distance()                                                                                                                                      
Out[3]: [0, 1]

In [4]: action_at_a_distance()                                                                                                                                      
Out[4]: [0, 1, 2]

In [5]: action_at_a_distance()                                                                                                                                      
Out[5]: [0, 1, 2, 3]

In [6]:         
```

Reply:
Wow, didn't see that coming!